libs/unity/library/Runtime/Scripts/Media/MicrophoneSource.cs

```diff
@@ -3,43 +3,99 @@
+    /// audio capture device (microphone). The audio source can be used to create one or more
+    /// audio tracks sharing the same audio content.
     /// </summary>
     [AddComponentMenu("MixedReality-WebRTC/Microphone Source")]
-    public class MicrophoneSource : AudioSender
+    public class MicrophoneSource : AudioTrackSource
     {
-        protected override async Task CreateLocalAudioTrackAsyncImpl()
+        public bool AutoGainControl => _autoGainControl;
+
+        [SerializeField]
+        [Tooltip("Enable automated gain control")]
+        [ToggleLeft]
+        protected bool _autoGainControl = true;
+
+#if UNITY_WSA && !UNITY_EDITOR
+        protected override async void OnEnable()
         {
-            if (Track == null)
+            // Request access to audio capture. The OS may show some popup dialog to the
+            // user to request permission. This will succeed only if the user approves it.
+            try
             {
-                // Ensure the track has a valid name
-                string trackName = TrackName;
-                if (trackName.Length == 0)
+                if (UnityEngine.WSA.Application.RunningOnUIThread())
+                {
+                    await RequestAccessAsync();
+                }
+                else
                 {
-                    trackName = Guid.NewGuid().ToString();
-                    // Re-assign the generated track name for consistency
-                    TrackName = trackName;
+                    UnityEngine.WSA.Application.InvokeOnUIThread(() => RequestAccessAsync(), waitUntilDone: true);
                 }
-                SdpTokenAttribute.Validate(trackName, allowEmpty: false);
+            }
+            catch (Exception ex)
+            {
+                // Log an error and prevent activation
+                Debug.LogError($"Audio access failure: {ex.Message}.");
+                this.enabled = false;
+                return;
+            }
 
+            // Once access was granted, continue to create the audio source by opening
+            // the audio capture device.
+            base.OnEnable();
+        }
+#endif
+
+        protected override async Task CreateAudioTrackSourceAsyncImpl()
+        {
+            if (Source == null)
+            {
                 // Create the local track
-                var trackSettings = new LocalAudioTrackSettings
+                var initConfig = new LocalAudioDeviceInitConfig
                 {
-                    trackName = trackName
+                    AutoGainControl = _autoGainControl,
                 };
-                Track = await LocalAudioTrack.CreateFromDeviceAsync(trackSettings);
-
-                // Synchronize the track status with the Unity component status
-                Track.Enabled = enabled;
+                Source = await WebRTC.AudioTrackSource.CreateFromDeviceAsync(initConfig);
             }
         }
```

Comment:
The flow of control going up in the base class and down into this class is more confusing than necessary. See other thread, it would be better to flatten `OnEnable` in the concrete class, maybe call into a helper function in `MediaTrackSource` doing the event raising.